library/src/main/java/com/dslplatform/json/JsonWriter.java

```diff
@@ -433,14 +424,22 @@ public final void writeAscii(final byte[] buf) {
 	 * @param len part of buffer to copy
 	 */
 	public final void writeAscii(final byte[] buf, final int len) {
+		writeAscii(buf, 0, len);
+	}
+
+	/**
+	 * Copy part of byte buffer into JSON as is.
+	 * Provided buffer can't be null.
+	 *
+	 * @param buf byte buffer to copy
+	 * @param offset in buffer to start from
+	 * @param len part of buffer to copy
+	 */
+	public final void writeAscii(final byte[] buf, final int offset, final int len) {
```

Comment:
When I was doing original writeAscii calls I was doing them on small buffers and actually considered to have a large array concatenated with all relevant info and call it with offsets, but in the end I didn't do that microoptimization.
But even for it it would use small ranges almost all the time so it would be better to call for loop instead of array copy.
Anyway,  its fine if this and the above method share code, but I would prefer it's a for loop.

Reply:
This will be suboptimal for my use case. I cache big responses (megs) in memory, and want to shovel them to clients asap, and for those arraycopy will be much better than looping over bytes. At the same for small arrays the extra call may be not worth it. What would you say to:

if len >= 1024:
  System.arraycopy
else:
  for loop

Reply:
Let's leave this APIs as is then and add a new one to reduce the confusion
writeRaw with offset and len which will use arraycopy

btw. I think you should be able to do this now as long as you are in the same namespace, but it's fine to have such method.

Reply:
Will open a separate PR then.